client/app/cloud/project/compute/loadbalancer/configure/compute-loadbalancer-configure.controller.js

```diff
@@ -0,0 +1,260 @@
+            .finally(() => self.loaders.loadbalancer = false);
+            $stateParams.validate = "";
+        }
+        else {
+            validatePromise = Promise.resolve("");
+        }
+        // After validation, load the loadbalancer
+        validatePromise.then(() => getLoadbalancer(true));
+
+        $scope.$on("$destroy", () => stopTaskPolling());
+        initGuides();
+    }
+
+    function initGuides() {
+        self.guides = {
+            horizon: ovhDocUrl.getDocUrl("g1773.creer_un_acces_a_horizon")
+        };
+    }
+
+    // Get servers of the default default farm of the frontend
+    function getAttachedServers() {
+        if (!self.loadbalancer.farm) {
+            return Promise.resolve([]);
+        }
+        return OvhApiIpLoadBalancing.Farm().Http().Server().Lexi().query({
+             serviceName : loadbalancerId,
+             farmId : self.loadbalancer.farm.farmId,
+        }).$promise
+        .then(serverIds =>
+            $q.all(
+                 _.map(serverIds, serverId => OvhApiIpLoadBalancing.Farm().Http().Server().Lexi().get({
+                         serviceName : loadbalancerId,
+                         farmId : self.loadbalancer.farm.farmId,
+                         serverId : serverId,
+                     }).$promise
+                 )
+            )
+        );
+    }
+
+    // Get cloud servers to add in the loadbalancer
+    function getServers() {
+        if (self.loaders.table.server) {
+            return;
+        }
+        self.loaders.table.server = true;
+        return $q.all({
+            cloudServers : OvhApiCloudProject.Instance().Lexi().query({ serviceName : serviceName }).$promise,
+            attachedServers : getAttachedServers(),
+        }).then(({cloudServers, attachedServers}) => {
+            self.attachedServers = {};
+            _.forEach(attachedServers, (attachedServer) => {
+                if(attachedServer.status == "active") {
+                    self.attachedServers[attachedServer.address] = attachedServer;
+                }
+            });
+            self.form.servers = _.mapValues(self.attachedServers, (e) => e ? true : false );
+            // Generate array of object type as {ipv4, name}
+            self.table.server =
+                _.flatten(_.map(cloudServers, (server) =>
+                    _.map(_.filter(server.ipAddresses, { type : "public", version : 4 }), (adresse) => ({ label : server.name, ip : adresse.ip }))
+                ));
+         }).catch((err) => {
+            self.table.server = null;
+            CloudMessage.error( [$translate.instant('cpc_server_error'), err.data && err.data.message || ''].join(' '))
+        }).finally(() => self.loaders.table.server = false);
+    }
+
+    // Configure and deploy the loadbalancer
+    self.configure = function () {
+        if (self.loaders.form.loadbalancer) {
+            return;
+        }
+        self.loaders.form.loadbalancer = true;
+        var promise = $q.resolve("");
```

Comment:
try using let and avoid using only promise in naming, also this function is very long and it's strange that there's no arguments

Reply:
I'm not sure you need to pass "" to resolve.